build.go

```diff
@@ -0,0 +1,192 @@
+package dotnetcoreaspnetruntime
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+	"time"
+
+	"github.com/paketo-buildpacks/packit/v2"
+	"github.com/paketo-buildpacks/packit/v2/chronos"
+	"github.com/paketo-buildpacks/packit/v2/postal"
+	"github.com/paketo-buildpacks/packit/v2/sbom"
+	"github.com/paketo-buildpacks/packit/v2/scribe"
+)
+
+//go:generate faux --interface EntryResolver --output fakes/entry_resolver.go
+type EntryResolver interface {
+	Resolve(string, []packit.BuildpackPlanEntry, []interface{}) (packit.BuildpackPlanEntry, []packit.BuildpackPlanEntry)
+	MergeLayerTypes(string, []packit.BuildpackPlanEntry) (launch, build bool)
+}
+
+//go:generate faux --interface DependencyManager --output fakes/dependency_manager.go
+type DependencyManager interface {
+	Deliver(dependency postal.Dependency, cnbPath, layerPath, platformPath string) error
+	GenerateBillOfMaterials(dependencies ...postal.Dependency) []packit.BOMEntry
+}
+
+//go:generate faux --interface VersionResolver --output fakes/version_resolver.go
+type VersionResolver interface {
+	Resolve(path string, entry packit.BuildpackPlanEntry, stack string) (postal.Dependency, error)
+}
+
+//go:generate faux --interface SBOMGenerator --output fakes/sbom_generator.go
+type SBOMGenerator interface {
+	GenerateFromDependency(dependency postal.Dependency, dir string) (sbom.SBOM, error)
+}
+
+//go:generate faux --interface ConfigParser --output fakes/config_parser.go
+type ConfigParser interface {
+	Parse(runtimeConfigFileGlob string) (string, error)
+}
+
+func Build(
+	entries EntryResolver,
+	dependencies DependencyManager,
+	versionResolver VersionResolver,
+	sbomGenerator SBOMGenerator,
+	configParser ConfigParser,
+	logger scribe.Emitter,
+	clock chronos.Clock,
+) packit.BuildFunc {
+	return func(context packit.BuildContext) (packit.BuildResult, error) {
+		logger.Title("%s %s", context.BuildpackInfo.Name, context.BuildpackInfo.Version)
+
+		// This no-op path is to accommodate for it the dotnet-publish buildpack
+		// generates a self-contained app
+		frameworkVersion, err := configParser.Parse(filepath.Join(context.WorkingDir, "*.runtimeconfig.json"))
+		if err != nil {
+			if errors.Is(err, os.ErrNotExist) {
+				logger.Process("Skipping build process as there was no runtimeconfig.json found")
+				logger.Break()
+				return packit.BuildResult{}, nil
+			}
+			return packit.BuildResult{}, err
+		}
+
+		if frameworkVersion == "" {
+			logger.Process("Skipping build process as there were no frameworks specified in the runtimeconfig.json")
```

Comment:
Would it be more clear for users if the logs said something like "Skipping build process because we think the app is self-contained"?

Reply:
I like that it adds the detail that it did the check on the runtimeconfig.json. How about, `Skipping build process, app detected as self-contained due to no frameworks in the runtimeconfig.json` ? Too wordy?